src/worker/mod.rs

```diff
@@ -197,6 +249,12 @@ fn worker_impl(display: Display, request_rx: Receiver<Command>, reply_tx: Sender
     // Setup sleep amount tracker.
     let mut sa_tracker = SleepAmountTracker::new(MAX_TIME_TO_SLEEP, MAX_WARM_WAKEUPS);
 
+    let (internal_tx, internal_rx) = channel();
+
+    std::thread::spawn(move || {
+        internal_worker_impl(internal_rx);
+    });
+
```

Comment:
I don't like that solution since we'll have one more thread permanently, so the smithay-clipboard will add 2 for no real reason. Using something like calloop would be more complicated, so given that this crate will be soon deprecated I'd be fine with on-demand thread creation, could notify that something is ready via condvar or something, but it shouldn't be too hard to make an ondemand worker instead of always keeping one around.